### DISCUSSION TITLE
JS-plugin & handling onSubmitAjax errors

### DISCUSSION THREAD

**Discussion by aatosm** [solspace/craft-freeform#157] [Questions & Answers] [answered]
Hi,

I would like to ask about Freeform js-plugin functionality related onSubmitAjax.

It seems like if onSubmitAjax request returns HTTP 400 (or anything else than 200), it just fails silently and the user is not really notified in any way. So my question would be, is there any way currently to handle the situation, where the onSubmitAjax fails?
At least events `freeform-ajax-error` or `freeform-ajax-after-submit` do not seem to trigger after the failed request.

Going through the plugin code, it seems like it works like this:
```
if (request.status === 200) {
  // succesful request ...
} else {
  console.log(request);
}
```

Many thanks!

**Comment by gustavs-gutmanis** [comment, marked as answer]
Hi @aatosm,

Thanks for bringing this up, we will update the code to better support this. By default, Craft doesn't use anything else than a `200` status code for ajax requests that aren't successful, but that doesn't mean we shouldn't support such cases.
